lib/ansible/modules/cloud/azure/azure_rm_dnsrecordset_info.py

```diff
@@ -221,7 +220,9 @@ def exec_module(self, **kwargs):
             results = self.list_zone()
 
         if is_old_facts:
-            self.results['ansible_facts']['azure_dnsrecordset'] = self.serialize_list(results)
+            self.results['ansible_facts'] = {
+                'azure_dnsrecordset': self.serialize_list(results)
+            }
         self.results['dnsrecordsets'] = self.curated_list(results)
```

Comment:
Shouldn't this only be returned for the non facts module?

Reply:
`dnsrecordsets` is the regular return value of the module; `ansible_facts.azure_dnsrecordset` is only returned when the module is called as `_facts` (i.e. when `is_old_facts` is `True`).

Reply:
It already returned that in `stable-2.8`: https://github.com/ansible/ansible/blob/stable-2.8/lib/ansible/modules/cloud/azure/azure_rm_dnsrecordset_facts.py#L212

(There were quite a lot of `_facts` modules which returned information also as non-facts, sometimes even just as non-facts. This apparently belongs to the former category.)

Reply:
Fair enough, I suppose this probably applies to most of my other comments below.

Reply:
It did, but not to all :)